tabular/tests/unittests/test_tabular.py

```diff
@@ -543,9 +484,107 @@ def get_benchmark_sets():
+            assert isinstance(y_pred_proba_oof_transformed, pd.Series)
+
+        assert y_pred_oof_transformed.equals(predictor.transform_labels(y_pred_oof, proba=False))
+
+        # Test that the transform_labels method is capable of reproducing the same output when converting back and forth, and test that oof 'transform' parameter works properly.
+        y_pred_proba_oof_inverse = predictor.transform_labels(y_pred_proba_oof, proba=True)
+        y_pred_proba_oof_inverse_inverse = predictor.transform_labels(y_pred_proba_oof_inverse, proba=True, inverse=True)
+        y_pred_oof_inverse = predictor.transform_labels(y_pred_oof)
+        y_pred_oof_inverse_inverse = predictor.transform_labels(y_pred_oof_inverse, inverse=True)
+
+        if isinstance(y_pred_proba_oof_transformed, pd.DataFrame):
+            pd.testing.assert_frame_equal(y_pred_proba_oof_transformed, y_pred_proba_oof_inverse)
+            pd.testing.assert_frame_equal(y_pred_proba_oof, y_pred_proba_oof_inverse_inverse)
+        else:
+            pd.testing.assert_series_equal(y_pred_proba_oof_transformed, y_pred_proba_oof_inverse)
+            pd.testing.assert_series_equal(y_pred_proba_oof, y_pred_proba_oof_inverse_inverse)
+        if isinstance(y_pred_oof_transformed, pd.DataFrame):
+            pd.testing.assert_frame_equal(y_pred_oof_transformed, y_pred_oof_inverse)
+            pd.testing.assert_frame_equal(y_pred_oof, y_pred_oof_inverse_inverse)
+        else:
+            pd.testing.assert_series_equal(y_pred_oof_transformed, y_pred_oof_inverse)
+            pd.testing.assert_series_equal(y_pred_oof, y_pred_oof_inverse_inverse)
+
+        # Test that index of both the internal training data and the oof outputs are consistent in their index values.
+        X_internal, y_internal = predictor.load_data_internal()
+        y_internal_index = list(y_internal.index)
+        assert list(X_internal.index) == y_internal_index
+        assert list(y_pred_oof.index) == y_internal_index
+        assert list(y_pred_proba_oof.index) == y_internal_index
+        assert list(y_pred_oof_transformed.index) == y_internal_index
+        assert list(y_pred_proba_oof_transformed.index) == y_internal_index
+    else:
+        # Raise exception
+        with pytest.raises(AssertionError):
+            predictor.predict_oof()
+        with pytest.raises(AssertionError):
+            predictor.predict_proba_oof()
+    if run_distill:
+        predictor.distill(time_limit=60, augment_args={"size_factor": 0.5})
+
+
+def run_tabular_benchmarks(fast_benchmark, subsample_size, perf_threshold, seed_val, fit_args, datasets=None, dataset_indices=None, run_distill=False, crash_in_oof=False, fit_helper=None):
     print("Running fit with args:")
     print(fit_args)
+
+    if fit_helper is not None:
+        if datasets is None:
+            datasets = [
+                "toy_binary_10",
+                "toy_multiclass_10",
+                "toy_regression_10",
+                "toy_quantile_10",
```

Comment:
Should there also be a softclass example included here?

Reply:
Yes, however none of our tests previously tested softclass as a standalone problem. Currently softclass is only used for distillation of multiclass tasks, and it isn't implemented in a way that works out of the box as a standalone problem type (I think, haven't checked). I'll look to add it in future.